src/test/java/tahafurkan/sandbox/usermanagement/service/UserServiceTest.java

```diff
@@ -92,4 +103,104 @@ void create_GivenAllowedUsername_ReturnsSavedUser() {
         Mockito.verify(userRepository, times(1)).save(user);
     }
 
+
+    @Test
+    void get_WithAnIdNotExist_ThrowsException() {
+        assertThrows(NoSuchUserExistsException.class, () -> {
+            int id = 1;
+            User user = new User();
+            user.setId(id);
+            userService.get(id);
+        });
+    }
+
+    @Test
+    void get_WithAnIdExists_ReturnUser() {
+        int id = 1;
+        User user = new User();
+        user.setId(id);
+        given(userRepository.findById(id)).willReturn(Optional.of(user));
+        userService.get(id);
+    }
+
+    @Test
+    void getAll_ReturnsEmptyUserList() {
+        userService.getAll();
+        Mockito.verify(userRepository, times(1)).findAll();
+    }
+
+    @Test
+    void getAll_ReturnsPresentUserList() {
+        List<User> records = new ArrayList<>();
+
+        List<Address> addressList = new ArrayList<>();
+        addressList.add(new Address(1, "home", "Istanbul", "Turkey"));
+        User user_1 = new User(1, "Taha", "Unsal", "taha.f.unsal@gmail.com","taha.furkan",  addressList);
+        User user_2 = new User(2, "Zehra", "Unsal", "zehra.unsal@gmail.com","zehra.unsal", addressList);
+        records.add(user_1);
+        records.add(user_2);
+
+        given(userRepository.findAll()).willReturn(records);
+
+        Mockito.verify(userRepository , verificationData -> records.size());
+        
+    }
+
+    @Test
+    void delete_WithNonExistentId_ThrowsException() {
+
+    }
```

Comment:
Asagidaki metoda benzer bir test olacak, yazamadigin kisim nedir?

Reply:
Bu test'te userRepository.deleteById() metoduna girmedigini de kontrol et. Ayrica userRepository.findById() metoduna girdigini de kontrol etmelisin. Bu dediğin kısmı anlamadım